minifi-toolkit/minifi-toolkit-configuration/src/main/java/org/apache/nifi/minifi/toolkit/configuration/ConfigMain.java

```diff
@@ -96,31 +102,39 @@ public int validate(String[] args) {
         }
         try (InputStream inputStream = pathInputStreamFactory.create(args[1])) {
             try {
-                ConfigSchema configSchema = SchemaLoader.loadConfigSchemaFromYaml(inputStream);
+                ConvertableSchema<ConfigSchema> configSchema = SchemaLoader.loadConvertableSchemaFromYaml(inputStream);
+                boolean valid = true;
                 if (!configSchema.isValid()) {
+                    System.out.println(FOUND_THE_FOLLOWING_ERRORS_WHEN_PARSING_THE_TEMPLATE_ACCORDING_TO_ITS_VERSION);
                     configSchema.getValidationIssues().forEach(s -> System.out.println(s));
                     System.out.println();
-                    return ERR_INVALID_CONFIG;
-                } else {
+                    valid = false;
+                    configSchema.clearValidationIssues();
+                }
+
+                ConfigSchema currentSchema = configSchema.convert();
```

Comment:
@JPercivall I'll take a stab at it. I'm not sure if it will be more or less confusing though, please let me know what you think.